src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -666,13 +670,21 @@
   (-plan->sql [query]
     (let [has-where? (boolean (:where query))
           has-projections? (not (empty? (:projected-fields query)))
-          update-when (fn [m pred ks f]
-                        (if pred
-                          (update-in m ks f)
-                          m))
+          update-hash-fields
```

Comment:
So you can't munge the hashes and uuids in the query-recs becaue they are `def`s and not `defn`s. Because they are `def`s clojure tries to evaluate the `sutils/postgres?` function before the db connection is available, at compile time, and gets a null pointer exception.